flake.nix

```diff
@@ -21,5 +21,14 @@
         formatter = pkgs.nixpkgs-fmt;
 
         devShells.default = import ./nix/shell.nix { inherit pkgs inputs; };
-      });
+      }) 
+    // rec {
+      overlays = {
+        default = overlays.yazi;
+        yazi = final: prev: {
+          yazi = final.callPackage ./nix/yazi.nix { };
```

Comment:
You can't re-use the `yazi` inside `eachDefaultSystem` because it's already instantiated into a specific architecture. An overlay should not be dependent on a specific architecture and accept whatever `nixpkgs` that was passed to it.